proxy/http2/Http2ClientSession.cc

```diff
@@ -183,7 +181,10 @@ Http2ClientSession::start()
 
   this->connection_state.init();
   send_connection_event(&this->connection_state, HTTP2_SESSION_EVENT_INIT, this);
-  this->handleEvent(VC_EVENT_READ_READY, read_vio);
+
+  if (this->sm_reader->is_read_avail_more_than(0)) {
```

Comment:
This implicitly says we have something to do if we can read 1 byte. What is it?

Reply:
May be Connection Preface ?

Reply:
Do we want the first byte of Connection Preface here? The length should be 10 bytes or so.

Reply:
I think we should not put these assumption here to make the logic simple.

Think about the error cases, if we start reading with 1 byte, the handler can reject the connection with malformed strings smaller than 10 bytes. ( actually, current `state_read_connection_preface()` doesn't do this  )

Also, HTTP/2 over TLS case (almost all the time) this is 0.

Reply:
Sounds like we could remove this since nobody cares cleartext H2. That would be more simpler.

Reply:
It's my first idea, but I'm not confident because there is a path of the buffer is given to the the `Http2SessionAccept` from the outside.